otter/cloud_client.py

```diff
@@ -135,10 +143,38 @@ def __init__(self, effect, tenant_id):
         self.tenant_id = tenant_id
 
 
+def _add_service_error_parsing(parser, request_func):
+    """
+    If a parser for API errors is found, use it to parse service API errors.
+
+    :param callable parser: A function which takes a tuple of exc info, to
+        be called when the request function raises an :class:`APIError`.
+    :param callable request_func: The request function to decorate
+
+    If there is no parser, does nothing.
+    """
+    if parser is None:
+        return request_func
+
+    def call_parser(excinfo):
+        # ensures that if the parser doesn't raise another exception,
+        # the original exception is raised
+        parser(excinfo)
+        six.reraise(*excinfo)
+
+    @wraps(request_func)
+    def request(*args, **kwargs):
+        return request_func(*args, **kwargs).on(
+            error=catch(APIError, call_parser))
+
+    return request
+
+
 def concretize_service_request(
         authenticator, log, service_configs,
         tenant_id,
-        service_request):
+        service_request,
+        service_error_parsers=None):
```

Comment:
I realize we discussed this on slack but passing parsers in this function seems like over-engineering to me. The intent here is to gradually provide specific exceptions so that existing callers do not break but eventually they would handle exceptions. So,
1. I don't see use-case to provide custom error parsers. Maybe I am missing something?
2. The new callers will now need to have (global?) parser dict and pass it here. 
I think a providing `specific_exceptions=True` argument should do the job.